### ISSUE TITLE
The conversion from the `varchar(255)` to `::text` produces the `null`

### ISSUE THREAD

**Issue by dadhi** [EnterpriseDB/mongo_fdw#133]
Using the release https://github.com/EnterpriseDB/mongo_fdw/releases/tag/REL-5_2_5
and
- Pg version `PostgreSQL 12.2 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39), 64-bit`
- Mongo 4.0.16

This does not work

```sql
SELECT varchar256_field::text 
FROM my_mongo_fdw_table; -- produces `null`
```

But if the same field listed twice - it works

```sql
SELECT 
varchar256_field 		as hack
varchar256_field::text  as text_value -- converts the actual value just fine
FROM my_mongo_fdw_table; -- produces `null`
```

It is not working though if you put it twice as a function argument

```sql
SELECT 
coalesce(varchar256_field, varchar256_field)::text -- does Not work
FROM my_mongo_fdw_table; -- produces `null`
```

**Comment by mjrussell** [comment]
I'm also hitting this issue in postgres 12. Interestingly, it appears to be an issue when converting from `json` implicitly or explicitly to any other type (`jsonb`,`text`, etc)

**Comment by sheim-dev** [comment]
@dadhi I had observed a similar bug when using mongo fdw <= 5.2.7 on postgres >= 10.0, seems this bug is fixed with release 5.2.8 of mongo fdw.
Can you confirm release 5.2.8 does indeed fix this issue for you?

**Comment by dadhi** [comment]
@sheim-dev I am no longer using mongo_fdw at the current project, so I won't be able to confirm. But the bug is easy reproducible, so I don't see a problem to check by the person who's using it.

**Comment by vaibhavdalvi93** [comment]
@dadhi, @sheim-dev We have fixed this issue in release 5.2.8.
Can you please let us know that this issue got resolved for you? If yes then will be closing this ticket.

**Comment by sheim-dev** [comment]
@vaibhavdalvi93 I can confirm release 5.2.8 does indeed fix the issues I had observed with queries incorrectly returning NULL on PostgreSQL versions >= 10 just like in this bug report.
From my perspective the issue can be closed, I'm also confident @dadhi original problem reported is fixed by release 5.2.8. Given he cannot provide independent explicit confirmation of that anymore, I think the issue should be closed.

**Comment by jeevanchalke** [comment]
Closing per @sheim-dev comment above.
